docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+# IBC Connection Handshake Spec
+
+> Work in progress.
+
+> All the content until section [2.2 Protocol] is stable and ready for review.
+
+> Reviews are welcome.
+
+
+## L1. Problem statement / outside view
+
+Suppose there are two chains (e.g., distributed ledgers) that execute independently of each other.
+What should be a basic abstraction for representing a path of communication between the two?
+We use the term _connection_ to represent this path of communication.
+The purpose of the IBC connection handshake protocol, at the highest level of abstraction, is to enable two chains to establish a connection with each other [[ICS 003](#references)].
```

Comment:
I think that it is more precisely to talk about connection establishment between two modules that are executing as deterministic state machines on different chains.

Reply:
I agree we should be precise and use these terms accurately.

I hesitated a lot between module, chain, and process. I ended up using "chain" as a compromise, but primarily motivated by two ideas: (1) connections are a per-chain abstraction (i.e., a chain executes the logic of establishing a connection; multiple modules  can share the same connection), and (2) the term _module_ is generic and overused (to an outsider reader, a "module" means not more than "subsystem" and so is quite vague; to an IBC expert, a [module](https://github.com/cosmos/ics/tree/master/spec/ics-024-host-requirements#module-system) is an implementation detail for separating different IBC functionalities).

It's worth noting that idea (1) is more complicated than what I wrote. It is very abstract to claim that a chain "executes the logic of establishing a connection". Concretely, this logic resides in a module. But this is just an implementation detail, just like the consensus logic of Tendermint Core resides in the "consensus reactor" (we don't usually speak about the reactor in high-level specs). This analogy to "reactors" was useful for me, but not sure how correct it is. What do you think?